omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -0,0 +1,1898 @@
+            orientation = np.array(reset_pose[1], dtype=float)[[3, 0, 1, 2]]
+            copy_robot.GetAttribute("xformOp:orient").Set(Gf.Quatd(*orientation)) 
+
+            robot_to_copy = None
+            if robot_type == "simplified":
+                robot_to_copy =  rc['robot']
+                og.sim.import_object(robot_to_copy)
+            else:
+                robot_to_copy = rc['robot']
+
+            # Copy robot meshes
+            for link in robot_to_copy.links.values():
+                link_name = link.prim_path.split("/")[-1]
+                for mesh_name, mesh in link.collision_meshes.items():
+                    split_path = mesh.prim_path.split("/")
+                    # Do not copy grasping frame (this is necessary for Tiago, but should be cleaned up in the future)
+                    if "grasping_frame" in link_name:
+                        continue
+
+                    copy_mesh_path = rc['copy_path'] + "/" + link_name
+                    copy_mesh_path += f"_{split_path[-1]}" if split_path[-1] != "collisions" else ""
+                    CopyPrimCommand(mesh.prim_path, path_to=copy_mesh_path).do()
+                    copy_mesh = get_prim_at_path(copy_mesh_path)
+                    relative_pose = T.relative_pose_transform(*mesh.get_position_orientation(), *link.get_position_orientation())
+                    relative_pose = (relative_pose[0], np.array([0, 0, 0, 1]))
+                    if link_name not in copy_robot_meshes.keys():
+                        copy_robot_meshes[link_name] = {mesh_name: copy_mesh}
+                        copy_robot_meshes_relative_poses[link_name] = {mesh_name: relative_pose}
+                    else:
+                        copy_robot_meshes[link_name][mesh_name] = copy_mesh
+                        copy_robot_meshes_relative_poses[link_name][mesh_name] = relative_pose
+
+                copy_robot_links_relative_poses[link_name] = T.relative_pose_transform(*link.get_position_orientation(), *robot.get_position_orientation())
+            
+            if robot_type == "simplified":
+                og.sim.remove_object(robot_to_copy)
+
+            robot_copy.prims[robot_type] = copy_robot
+            robot_copy.meshes[robot_type] = copy_robot_meshes
+            robot_copy.relative_poses[robot_type] = copy_robot_meshes_relative_poses
+            robot_copy.links_relative_poses[robot_type] = copy_robot_links_relative_poses
+
+        og.sim.step()
+        return robot_copy
+
+    def get_action_space(self):
+        if ACTIVITY_RELEVANT_OBJECTS_ONLY:
+            assert isinstance(self.task, BehaviorTask), "Activity relevant objects can only be used for BEHAVIOR tasks"
+            self.addressable_objects = sorted(set(self.task.object_scope.values()), key=lambda obj: obj.name)
+        else:
+            self.addressable_objects = sorted(set(self.scene.objects_by_name.values()), key=lambda obj: obj.name)
```

Comment:
in this case, the objects can change tho, right?

Reply:
or, rather, the number of objects

Reply:
Unsure how exactly this will work in the context of changing object counts. I'll leave a TODO here.